src/items.rs

```diff
@@ -1828,6 +1825,60 @@ fn is_empty_infer(ty: &ast::Ty, pat_span: Span) -> bool {
     }
 }
 
+/// Recover any missing comments between the argument and the type.
+///
+/// # Returns
+///
+/// A 2-len tuple with the comment before the colon in first position, and the comment after the
+/// colon in second position.
+fn get_missing_arg_comments(
+    context: &RewriteContext<'_>,
+    pat_span: Span,
+    ty_span: Span,
+    shape: Shape,
+) -> (String, String) {
+    let missing_comment_span = mk_sp(pat_span.hi(), ty_span.lo());
+
+    let span_before_colon = {
+        let missing_comment_span_hi = context
+            .snippet_provider
+            .span_before(missing_comment_span, ":");
+        mk_sp(pat_span.hi(), missing_comment_span_hi)
+    };
+    let span_after_colon = {
+        let missing_comment_span_lo = context
+            .snippet_provider
+            .span_after(missing_comment_span, ":");
+        mk_sp(missing_comment_span_lo, ty_span.lo())
+    };
+
+    let comment_before_colon = rewrite_missing_comment(span_before_colon, shape, context);
+    let comment_after_colon = rewrite_missing_comment(span_after_colon, shape, context);
+    let leading_space = |com: String| {
+        if com.is_empty() {
+            String::new()
+        } else {
+            format!(" {}", com)
+        }
+    };
+    let trailing_space = |com: String| {
+        if com.is_empty() {
+            String::new()
+        } else {
+            format!("{} ", com)
+        }
+    };
+    match (comment_before_colon, comment_after_colon) {
```

Comment:
It's more or less a matter of a taste...however I would rather write these as something like the following:

```rust
comment_before_colon.map(leading_space).unwrap_or(String::new)
```

You may need to change the signature of closures to `|com: &str|`.

Reply:
Thanks for the suggestion, I like it! I tweaked it a bit but I kept the spirit.

Reply:
Nice, even better :)